src/http/common.cc

```diff
@@ -169,7 +169,16 @@ class http_content_length_data_sink_impl : public data_sink_impl {
         _bytes_written = 0;
     }
     virtual future<> put(net::packet data) override {
-        return data_sink_impl::fallback_put(std::move(data));
+        auto size = data.len();
+        if (size == 0 || _bytes_written == _limit) {
+            return make_ready_future<>();
+        }
+        if (_bytes_written + size > _limit) {
+            return make_exception_future<>(std::runtime_error(format("body conent length overflow: want {} limit {}", _bytes_written + size, _limit)));
+        }
+        return _out.write(std::move(data)).then([this, size] {
+            _bytes_written += size;
```

Comment:
Better to adjust _bytes_written before the write, saves a continuation. If write() throws, we won't look at _bytes_written anyway.

Ah, you need to add the exact amount written in case the remote closed the connection during the transfer. And we can'd return a future carrying a size anyway. Oh well.